cmd/crdvalidator/internal/crd/crd.go

```diff
@@ -0,0 +1,172 @@
+package crd
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/validation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func Validate(ctx context.Context, cl client.Client, newCrd *apiextensionsv1.CustomResourceDefinition) error {
```

Comment:
I think it would be helpful to have a comment here that explains what the validation does. It seems pretty involved. So having this upfront might help when reading the code for the first time (even if after a long time)

Reply:
Agreed, I'll add something more descriptive.